Source: https://github.com/lxde/pcmanfm/pull/17/commits/d0b880a081adaf0d301824ca3db3f05090a992e3

Fix build with gcc14 -Werror=incompatible-pointer-types
gcc14 now defaults to -Werror=incompatible-pointer-types.

Add required casts to the returned pointers.

--- pcmanfm-1.3.2-origin/src/desktop.c
+++ pcmanfm-1.3.2/src/desktop.c
@@ -1267,7 +1267,7 @@
         item = items->data;
         if (item->item->is_selected)
             if (i-- == 0)
-                return g_object_ref(item);
+                return (AtkObject *)g_object_ref(item);
     }
     return NULL;
 }
@@ -1454,7 +1454,7 @@
     item = g_list_nth_data(priv->items, index);
     if (!item)
         return NULL;
-    return g_object_ref(item);
+    return (AtkObject *)g_object_ref(item);
 }
 
 static void fm_desktop_accessible_initialize(AtkObject *accessible, gpointer data)
